Daf.Meta.Editor/ViewModels/DataSourceViewModel.cs

```diff
@@ -0,0 +1,247 @@
+﻿// SPDX-License-Identifier: MIT
+// Copyright © 2021 Oscar Björhn, Petter Löfgren and contributors
+
+using System.Collections.ObjectModel;
+using System.ComponentModel.DataAnnotations;
+using Daf.Meta.Layers;
+using Microsoft.Toolkit.Mvvm.ComponentModel;
+using Microsoft.Toolkit.Mvvm.Messaging;
+using PropertyTools.DataAnnotations;
+
+namespace Daf.Meta.Editor.ViewModels
+{
+	public abstract class DataSourceViewModel : ObservableValidator
+	{
+		protected DataSourceViewModel(DataSource dataSource)
+		{
+			DataSource = dataSource;
+
+			// Forces the binding for ColumnsNotInHubsOrLinks property to update. This needs to happen both when a StagingColumn is added/removed and when a HubMapping is changed.
+			WeakReferenceMessenger.Default.Register<DataSourceViewModel, HubRelationshipChanged>(this, (r, m) =>
+			{
+				OnPropertyChanged(nameof(ColumnsNotInHubsOrLinks));
+			});
+
+			WeakReferenceMessenger.Default.Register<DataSourceViewModel, StagingColumnAddedRemoved>(this, (r, m) => OnPropertyChanged(nameof(ColumnsNotInHubsOrLinks)));
+		}
+
+		[Browsable(false)]
+		public virtual DataSource DataSource { get; }
+
+		public ObservableCollection<Satellite> Satellites => DataSource.Satellites;
+
+		public StagingTable? StagingTable => DataSource.StagingTable;
+
+		[Category("General")]
+		[Description("The name of the data source")]
+		public string Name
+		{
+			get => DataSource.Name;
+			set
+			{
+				SetProperty(DataSource.Name, value, DataSource, (dataSource, name) => dataSource.Name = name, true);
+			}
+		}
+
+		// Can't find any references to this.
```

Comment:
What does this mean?

Reply:
I can't find any view that displays this property, so I'm unsure if it even needs to be in the view model. Can you say if it can be safely removed?